keps/sig-node/4622-topologymanager-max-allowable-numa-nodes/README.md

```diff
@@ -0,0 +1,757 @@
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+We propose to add a new `TopologyManager` policy option called `max-allowable-numa-nodes` to the static TopologyManager policy. It can configure the value of maxAllowableNUMANodes in the TopologyManager. the current hard-coded value of 8 was added as a stop-gap 4 years ago, a configurable option policy option can improve the topology manager to remove the state explosion that occurs when trying to enumerate the possible NUMA affinities and generating their hints.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As developer in the AI space, I want to use AI accelerators "super chips" which expose ARM cores with more than 8 NUMA nodes.  
+
+#### Story 2
+
+As user in the high-performance-computing space, I want to enable the sub-NUMA or NUMA-per-socket option of my high-end x86 CPU, which will bring  
+the count of NUMA nodes to exceed 8.  
+
+#### Story 3
+
+As administrator of edge nodes, I want to use power-efficient yet massively parallel ARM chips which expose more 
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+Setting values higher than the current default may cause performance degradation on admission time, because this is the flow on which the Topology Manager operates. Fixing this will require rearchitecting the Topology Manager and it is thus out of scope of this KEP.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+The risk associated with implementing this new proposal is minimal. It pertains only to a distinct policy option within the `TopologyManager` and is safeguarded by the option's inherent security measures, in addition to the default deactivation of the `TopologyManagerPolicyBetaOptions` feature gate.
+
+| Risk                                             | Impact | Mitigation |
+| -------------------------------------------------| -------| ---------- |
+| Set a value lower 8 causes kubelet crash         | High   | the minimum value legal value should be the current hardcoded value(8), If not, we should log it and fail |
```

Comment:
I don't understand this. Setting the value lower should be no problem. It just means the kubelet will fail if their actual machine has more NUMA nodes than is specified.
```suggestion
| Set a value lower 8 causes kubelet crash         | High   | the minimum value legal value should be the current hardcoded value(8), If not, we should log it and fail |
```

Reply:
This was a suggestion of mine. Albeit legal, I see no value in restricting the maximum NUMA nodes the kubelet can deal with. Furthermore, it will be inconsistent with how cadvisor reports the data about the machine. If we had a common central way to expose a subset of the machine HW, it would have been different, but in this case I see only a source of confusion.

That said if you think differently and prefer a more GIGO approach I'm open to change direction.

Reply:
OK; I'm fine with leaving it. The only reason someone might want to reduce it is if their platform is so resource constrained that even calculating with 8 NUMA nodes is problematic. However, if their node is so resource constrained in the first place, then it probably won't actually be that slow (because there wont be many devices it needs to do calculations across).

Reply:
I will contiune keep this here :)